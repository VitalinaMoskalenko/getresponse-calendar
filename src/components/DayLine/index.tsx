import React from "react";
import styled from "styled-components";
import DayItem from "../DayItem";
import { H3 } from "../Headings";
import { SmallBody } from "../Paragraphs";

type ContainerPropsType = {
  isSelected: boolean;
  isGuiltFreeDay?: boolean;
};

const Container = styled.div<ContainerPropsType>`
  display: flex;
  flex-direction: column;
  justify-content: ${({ isGuiltFreeDay }) => isGuiltFreeDay && "space-between"};
  width: fit-content;
  border: ${({ isSelected, theme }) =>
    isSelected ? `2px solid ${theme.colors.orange}` : "0px"};
  margin-top: ${({ isSelected }) => (isSelected ? "0px" : "2px")};
`;

const DayNumber = styled.div<ContainerPropsType>`
  display: flex;
  justify-content: center;
  width: 100px;
  height: 40px;
  padding: 10px;
  border-bottom: 2px solid ${({ theme }) => theme.colors.darkGrey};
  color: ${({ isSelected, theme }) =>
    isSelected ? `${theme.colors.orange}` : `${theme.colors.green}`};
`;

const AdditionalContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
`;

const CarbsContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100px;
  height: 20px;
  padding: 10px;
  background-color: ${({ theme }) => theme.colors.lightGrey};
  border-bottom: 2px solid ${({ theme }) => theme.colors.darkGrey};
`;

const WorkoutContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100px;
  height: 20px;
  padding: 10px;
  background-color: ${({ theme }) => theme.colors.lightGrey};
`;

const PrintContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100px;
  margin-bottom: 2px;
  height: 62px;
  padding: 10px;
  border-top: 2px solid ${({ theme }) => theme.colors.darkGrey};
  background-color: ${({ theme }) => theme.colors.lightGrey};
`;

const GuiltFreeDayContainer = styled.div`
  align-items: center;
  display: flex;
  writing-mode: vertical-lr;
  transform: rotate(180deg);
  color: ${({ theme }) => theme.colors.darkGrey};
`;

const IconWorkout = styled.img``;
const IconPrinter = styled.img`
  width: 50px;
`;

type PropsType = {
  title?: string;
  dayNumber: number;
  carbs?: string;
  workout?: boolean;
  guiltFreeDay?: boolean;
  day?: {
    name: string;
    isAccomplished: boolean;
  }[];
};

const DayLine = ({
  title,
  dayNumber,
  day,
  guiltFreeDay,
  carbs,
  workout,
}: PropsType) => {
  return (
    <Container isGuiltFreeDay={guiltFreeDay} isSelected={dayNumber === 65}>
      <DayNumber isSelected={dayNumber === 65}>
        <H3>{title}</H3>
      </DayNumber>
      {guiltFreeDay ? (
        <GuiltFreeDayContainer>
          <H3>GUILT-FREE DAY</H3>
        </GuiltFreeDayContainer>
      ) : (
        day?.map((item, index) => (
          <DayItem
            key={index}
            meal={item.name}
            isAccomplished={item.isAccomplished}
          />
        ))
      )}
      {guiltFreeDay ? (
        <PrintContainer>
          <IconPrinter src="https://cdn2.iconfinder.com/data/icons/ios-7-icons/50/print-512.png" />
        </PrintContainer>
      ) : (
        <AdditionalContainer>
          <CarbsContainer>
            <SmallBody>{carbs}</SmallBody>
          </CarbsContainer>
          <WorkoutContainer>
            <IconWorkout />
          </WorkoutContainer>
        </AdditionalContainer>
      )}
    </Container>
  );
};

export default DayLine;
