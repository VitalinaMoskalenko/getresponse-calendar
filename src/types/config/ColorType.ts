export enum ColorType {
  white = "white",
  black = "black",
  lightGrey = "#e1e1e1",
  darkGrey = "#535353",
  green = "#b2c200",
  orange = "#ff801a",
}
