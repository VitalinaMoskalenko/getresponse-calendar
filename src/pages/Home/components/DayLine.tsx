import React from "react";
import styled from "styled-components";
import { SmallBody } from "../../../components/Paragraphs";

const Container = styled.div`
  display: flex;
  flex: 1;
  flex-direction: row;
  justify-content: space-between;
  align-items: stretch;
`;

const TimeBox = styled.div`
  background-color: aqua;
`;
const EmptyBox = styled.div`
  height: 40px;
  width: 80px;
  background-color: aqua;
`;

const Workout = styled(SmallBody)`
  align-self: flex-end;
`;

const DayLine = () => {
  return (
    <Container>
      <TimeBox>
        <SmallBody>6:00AM</SmallBody>
      </TimeBox>
      <TimeBox>
        <SmallBody>9:00AM</SmallBody>
      </TimeBox>
      <TimeBox>
        <SmallBody>12:00PM</SmallBody>
      </TimeBox>
      <TimeBox>
        <SmallBody>3:00PM</SmallBody>
      </TimeBox>
      <TimeBox>
        <SmallBody>6:00PM</SmallBody>
      </TimeBox>
    </Container>
  );
};

export default DayLine;
