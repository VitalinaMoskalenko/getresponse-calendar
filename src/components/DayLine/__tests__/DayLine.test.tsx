import React from "react";
import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";
import DayLine from "../";
import { theme } from "../../../config";

it("renders DayLine correctly", () => {
  const dayLineComponent = renderer
    .create(
      <ThemeProvider theme={theme}>
        <DayLine dayNumber={65} day={[{ name: "", isAccomplished: false }]} />
      </ThemeProvider>
    )
    .toJSON();

  expect(dayLineComponent).toMatchSnapshot();
});
