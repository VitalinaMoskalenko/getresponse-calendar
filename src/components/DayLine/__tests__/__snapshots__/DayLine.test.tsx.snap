// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders DayLine correctly 1`] = `
<div
  className="sc-pNWdM fPjrWe"
>
  <div
    className="sc-jrsJWt gqHGaR"
  >
    <h1
      className="sc-gKAaRy kalXHB"
    />
  </div>
  <div
    className="sc-dlnjwi bfyrKl"
  >
    <p
      className="sc-gtsrHT fsyTPD"
    >
      
    </p>
  </div>
  <div
    className="sc-kEqXSa bTSwWp"
  >
    <div
      className="sc-iqAclL vUoeS"
    >
      <p
        className="sc-gtsrHT fsyTPD"
      />
    </div>
    <div
      className="sc-crzoAE gZtnyw"
    >
      <img
        className="sc-ksluID"
      />
    </div>
  </div>
</div>
`;
